### ISSUE TITLE
X-Ray engine source code available on GitHub

### ISSUE THREAD

**Issue by Anthony-Gaudino** [OpenTechEngine/Discussions#17]
I was reading about the X-Ray engine used on the Stalker games and I found out that the source code is available on GitHub. I'm not sure if it's a leaked source or the developers released it, `it's not free software` as the license file tells.

You can find it here: https://github.com/OpenXRay/xray-16

I got surprised as I didn't see much information about the source release, that's why I think it may be leaked source.

Anyway, I think it's OK if one study the code.

**Comment by kortemik** [comment]
The code release seems intended, however their license is not compatible with ours and I wonder what is their intention with releasing the source under such license. Perhaps to fish developers on their engine but with a license like that it's not going to fly too far..

**Comment by BielBdeLuna** [comment]
this is excellent news, xray includes a planer, this could serve as inspiration for developing a planer for the AI,. this and also FEAR sdk expose several examples of a planer. a GOAP system (Goal Oriented Action Planer)

**Comment by ghost** [comment]
@BielBdeLuna GOAP would be great, especially for military/tactical squad simulations. Imagine OTE with machine learning applications to learn, study and simulate strategy for real world application. The security industry would appreciate that.

**Comment by BielBdeLuna** [comment]
technically GOAP can be used for everything not just tactical action, imagine GOAP as an automatized state machine, so instead of you doing the state machine states, the computer computes it for you.

**Comment by ghost** [comment]
That would be great, also for wildlife and small random weather events.

**Comment by BielBdeLuna** [comment]
no, the problem of GOAP is that per tick it is calculating and planning all possible plans and scoring it. so it needs a lot of processing power instead of the static FSM which only process the actual orders given by the script.

for random events and wildlife there should be a less expensive method

**Comment by ghost** [comment]
@BielBdeLuna Would calculating possible outcomes and storing it into a cache folder perhaps help improve performance? The only reason I'm thinking of it is after testing some performance differences with the shader cache written by the Nvidia drivers to take some load off the GPU memory.

**Comment by DanielGibson** [comment]
> Imagine OTE with machine learning applications to learn, study and simulate strategy for real world application. The security industry would appreciate that.

I'd prefer if that scum wouldn't use my code :-P

**Comment by ghost** [comment]
@DanielGibson I know how you feel, Rage against the machine. Lately it seems that policing and security is leaning towards either the fascist or mobster sides.

**Comment by BielBdeLuna** [comment]
no, within a dynamic world, the evaluation of every plan could change per tick.

**Comment by ForserX** [comment]
If someone is interested in XRay Engine, then you are welcome: https://github.com/xrOxygen/xray-oxygen
